### PR TITLE
refactor: migrate api key module to DML

### PR DESCRIPTION
**Pull request by thetutlage**
Fixes: FRMW-2827

### CODE REVIEW

File: packages/modules/api-key/src/migrations/Migration20241205122700.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export class Migration20241205122700 extends Migration {
+  async up(): Promise<void> {
+    this.addSql(
+      'alter table if exists "api_key" add column if not exists "deleted_at" timestamptz null;'
+    )
+    this.addSql(
+      'alter table if exists "api_key" alter column "type" type text using ("type"::text);'
+    )
+    this.addSql(
+      'alter table if exists "api_key" add constraint "api_key_type_check" check ("type" in (\'publishable\', \'secret\'));'
```
Comment (olivermrbl) [inline packages/modules/api-key/src/migrations/Migration20241205122700.ts:12]: question: Are these constraints autogenerated from the enum property in the DML?

File: packages/modules/api-key/src/migrations/Migration20241205122700.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { Migration } from "@mikro-orm/migrations"
+
+export class Migration20241205122700 extends Migration {
+  async up(): Promise<void> {
+    this.addSql(
+      'alter table if exists "api_key" add column if not exists "deleted_at" timestamptz null;'
+    )
+    this.addSql(
+      'alter table if exists "api_key" alter column "type" type text using ("type"::text);'
```
Comment (olivermrbl) [inline packages/modules/api-key/src/migrations/Migration20241205122700.ts:9]: question: Looks like this is redundant, since it's part of the up and down?

**Review by olivermrbl — COMMENTED** [review]
LGTM

File: packages/modules/api-key/src/migrations/Migration20241205122700.ts (hunk 1 above)
Comment (thetutlage) [reply packages/modules/api-key/src/migrations/Migration20241205122700.ts:12]: Yes

Comment (olivermrbl) [reply packages/modules/api-key/src/migrations/Migration20241205122700.ts:12]: Neat

File: packages/modules/api-key/src/migrations/Migration20241205122700.ts (hunk 2 above)
Comment (thetutlage) [reply packages/modules/api-key/src/migrations/Migration20241205122700.ts:9]: Yes, looks redundant, but again generated automatically by MikroORM after switching to DML

**Review by adrien2p — APPROVED** [review]

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Pull request merged**